### PR TITLE
adds label enforcement for PRs

### LINKED ISSUE #6311: CI should fail if the PR does not have a canary/beta or release label

### PR DESCRIPTION
**Pull request by runspired**
resolves #6311

assists #6103
assists #7555 
assists #6894 

This PR aims to enforce that

- PRs are explicitly labeled for the changelog (or to skip the changelog)
- PRs are explicitly labeled for back porting needs (or whether they are the needed backport PR)
- that labels that shouldn't be applied within a context are not applied.

We will need to update the lts and lts-prev branch names when new lts versions occur.

### CODE REVIEW

File: .github/workflows/enforce-pr-labels-release.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+name: Enforce Release PR labeling
+
+on:
+  pull_request:
+    types: [labeled, unlabeled, opened, edited, synchronize]
```
Comment (snewcomer) [inline .github/workflows/enforce-pr-labels-release.yml:5]: Do you happen to know what a "synchronized" pull request is?  See any other ones we would want to add from the docs below?

https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request

**Review by snewcomer — APPROVED** [review]
Would be nice if they allowed a "common-" yml file that has overrided "on" steps for each branch that we target.  Otherwise, this seems like a great idea!

File: .github/workflows/enforce-pr-labels-release.yml (hunk 1 above)
Comment (runspired) [reply .github/workflows/enforce-pr-labels-release.yml:5]: I believe it means that the PR is from a fork and the fork was updated. I went through those docs and this is the full list I think we need.

Comment (runspired) [reply .github/workflows/enforce-pr-labels-release.yml:5]: oooh, other than `reopened`, which I added to some but apparently not all. Will make sure all have it.

**Pull request merged**